rust/cli/src/commands/vacuum.rs

```diff
@@ -0,0 +1,289 @@
+use crate::utils::{
+    get_frontend_config, LocalFrontendCommandArgs, DEFAULT_PERSISTENT_PATH, SQLITE_FILENAME,
+};
+use chroma_config::registry::Registry;
+use chroma_config::Configurable;
+use chroma_frontend::frontend::Frontend;
+use chroma_frontend::FrontendConfig;
+use chroma_log::Log;
+use chroma_segment::local_segment_manager::LocalSegmentManager;
+use chroma_sqlite::db::SqliteDb;
+use chroma_sysdb::SysDb;
+use chroma_system::System;
+use chroma_types::{CollectionUuid, ListCollectionsRequest};
+use clap::Parser;
+use colored::Colorize;
+use dialoguer::Confirm;
+use indicatif::{ProgressBar, ProgressStyle};
+use serde_json::json;
+use sqlx::Row;
+use std::error::Error;
+use std::path::Path;
+use std::str::FromStr;
+use std::{fs, io};
+
+#[derive(Parser, Debug)]
+pub struct VacuumArgs {
+    #[clap(flatten)]
+    frontend_args: LocalFrontendCommandArgs,
+    #[arg(long)]
+    force: bool,
+}
+
+fn sizeof_fmt(num: u64, suffix: Option<&str>) -> String {
+    let suffix = suffix.unwrap_or("B");
+    let mut n = num as f64;
+    let units = ["", "Ki", "Mi", "Gi", "Ti", "Pi", "Ei", "Zi"];
+    for unit in &units {
+        if n.abs() < 1024.0 {
+            return format!("{:3.1}{}{}", n, unit, suffix);
+        }
+        n /= 1024.0;
+    }
+    format!("{:.1}Yi{}", n, suffix)
+}
+
+fn get_dir_size(path: &Path) -> io::Result<u64> {
+    let mut total_size = 0;
+    for entry in fs::read_dir(path)? {
+        let entry = entry?;
+        let entry_path = entry.path();
+        let metadata = fs::metadata(&entry_path)?;
+        if metadata.is_file() {
+            total_size += metadata.len();
+        } else if metadata.is_dir() {
+            total_size += get_dir_size(&entry_path)?;
+        }
+    }
+    Ok(total_size)
+}
+
+async fn trigger_vector_segments_max_seq_id_migration(
+    sqlite: &SqliteDb,
+    sysdb: &mut SysDb,
+    segment_manager: &LocalSegmentManager,
+) -> Result<(), Box<dyn Error>> {
+    let rows = sqlx::query(
+        r#"
+                SELECT collection FROM "segments"
+                WHERE "id" NOT IN (SELECT "segment_id" FROM "max_seq_id") AND "type" = 'urn:chroma:segment/vector/hnsw-local-persisted'
+            "#,
+    ).fetch_all(sqlite.get_conn()).await?;
+
+    let collection_ids: Result<Vec<CollectionUuid>, _> = rows
+        .into_iter()
+        .map(|row| CollectionUuid::from_str(row.get::<&str, _>(0)))
+        .collect();
+
+    let collection_ids = collection_ids?;
+
+    for collection_id in collection_ids {
+        let collection = sysdb.get_collection_with_segments(collection_id).await?;
+
+        // If collection is uninitialized, that means nothing has been written yet.
+        let dim = match collection.collection.dimension {
+            Some(dim) => dim,
+            None => continue,
+        };
+
+        segment_manager
+            .get_hnsw_writer(&collection.vector_segment, dim as usize)
```

Comment:
let's add a Rust test or update the existing Python test and make sure this codepath gets triggered correctly